driver/src/solution_submission/mod.rs

```diff
@@ -120,48 +128,189 @@ impl<'a> StableXSolutionSubmitting for StableXSolutionSubmitter<'a> {
         solution: Solution,
         claimed_objective_value: U256,
     ) -> Result<(), SolutionSubmissionError> {
-        self.contract
-            .submit_solution(
-                batch_index.clone(),
-                solution.clone(),
-                claimed_objective_value,
+        retry_with_gas_price_increase(
+            self.contract,
+            batch_index,
+            solution.clone(),
+            claimed_objective_value,
+            self.gas_price_estimating,
+            60_000_000_000u64.into(),
+        )
+        .map_err(|err| {
+            extract_transaction_receipt(&err)
+                .and_then(|tx| {
+                    let block_number = tx.block_number?;
+                    match self.contract.get_solution_objective_value(
```

Comment:
What is the purpose of this?

Reply:
This was there before, the diff is just struggling to show it correctly. It was used to get the revert message in case of a transaction failure.